DataDog/datadog-lambda-python
# cold start tracing (python)

### What does this PR do?

Add **Cold Start Tracing** feature to Datadog lambda python layer.
This feature is already available in `datadog-lambda-js` layer and this is [its PR link](https://github.com/DataDog/datadog-lambda-js/pull/336).
### Motivation

Showing the spans to indicate the time consumed by loading each modules can help users better understand the cold start causes.

### Testing Guidelines

Deployed in a testing lambda. 
1. Check the spans show up as expected. <img width="1128" alt="Screenshot 2023-02-13 at 10 14 30 AM" src="https://user-images.githubusercontent.com/5253430/218496718-75679541-ca7e-4512-b273-8b4f8612efff.png">
2. Invoke it multiple times to see non-cold-start cases.<img width="1135" alt="Screenshot 2023-02-13 at 10 18 02 AM" src="https://user-images.githubusercontent.com/5253430/218497381-069fefd5-2dae-4018-a019-046905332f98.png">
3. Tested with the [inferred-spans-python setup](https://github.com/DataDog/cloud-integrations-lambdas/blob/master/inferred-spans-python/README.md), which has one lambda triggering following lambdas via SQS/SNS etc. <img width="1126" alt="Screenshot 2023-02-13 at 3 34 44 PM" src="https://user-images.githubusercontent.com/5253430/218568825-ec2d66a5-1a76-4748-8e34-429030467649.png">
4. Change the env variables to see they are working as expected. 
   - with `DD_COLD_START_TRACING`=`false` <img width="1116" alt="Screenshot 2023-02-13 at 10 20 47 AM" src="https://user-images.githubusercontent.com/5253430/218498940-186cbbb8-62c5-4fad-8cf3-0a26f1b65db9.png">
   - with `DD_MIN_COLD_START_DURATION`=`80`<img width="1122" alt="Screenshot 2023-02-13 at 10 26 37 AM" src="https://user-images.githubusercontent.com/5253430/218500087-b35e07cf-b7e3-439a-ad96-ef311091659f.png">
   - with `DD_COLD_START_TRACE_SKIP_LIB`=`ddtrace.internal.sampling,jsonschema`<img width="1119" alt="Screenshot 2023-02-13 at 10 27 56 AM" src="https://user-images.githubusercontent.com/5253430/218500341-bd9565db-b731-40ff-9cbf-894ba743e8f4.png">





### Additional Notes



### Types of Changes


### Check all that apply
